test/integration/global_deploy_test.rb

```diff
@@ -116,4 +126,54 @@ def test_global_deploy_task_failure
       "Template validation failed",
     ])
   end
+
+  def test_global_deploy_prune_black_box_success
+    assert_deploy_success(deploy_global_fixtures('globals', clean_up: false, selector: 'test=prune1'))
+    reset_logger
+    assert_deploy_success(deploy_global_fixtures('globals', subset: 'storage_classes.yml', selector: 'test=prune1'))
+    assert_logs_match_all([
+      "Phase 1: Initializing deploy",
+      "Using resource selector test=prune1",
+      "All required parameters and files are present",
+      "Discovering resources:",
+      "  - StorageClass/testing-storage-class",
+      "Phase 2: Checking initial resource statuses",
+      %r{StorageClass\/testing-storage-class[\w-]+\s+Exists},
+      "Phase 3: Deploying all resources",
+      %r{Deploying StorageClass\/testing-storage-class[\w-]+ \(timeout: 300s\)},
+      "Don't know how to monitor resources of type StorageClass.",
+      %r{Assuming StorageClass\/testing-storage-class[\w-]+ deployed successfully.},
+      %r{Successfully deployed in [\d.]+s: StorageClass\/testing-storage-class},
+      "Result: SUCCESS",
+      "Pruned 1 resource and successfully deployed 1 resource",
+      "Successful resources",
+      "StorageClass/testing-storage-class",
+    ])
+  end
+
+  def test_no_prune_global_deploy_black_box_success
+    assert_deploy_success(deploy_global_fixtures('globals', clean_up: false, selector: 'test=prune2'))
+    reset_logger
+    assert_deploy_success(deploy_global_fixtures('globals', subset: 'storage_classes.yml',
+      selector: "test=prune2", prune: false))
+    assert_logs_match_all([
+      "Phase 1: Initializing deploy",
+      "Using resource selector test=prune2",
+      "All required parameters and files are present",
+      "Discovering resources:",
+      "  - StorageClass/testing-storage-class",
+      "Phase 2: Checking initial resource statuses",
+      %r{StorageClass\/testing-storage-class[\w-]+\s+Exists},
+      "Phase 3: Deploying all resources",
+      %r{Deploying StorageClass\/testing-storage-class[\w-]+ \(timeout: 300s\)},
+      "Don't know how to monitor resources of type StorageClass.",
+      %r{Assuming StorageClass\/testing-storage-class[\w-]+ deployed successfully.},
+      %r{Successfully deployed in [\d.]+s: StorageClass\/testing-storage-class},
+      "Result: SUCCESS",
+      "Successfully deployed 1 resource",
+      "Successful resources",
+      "StorageClass/testing-storage-class",
+    ])
+    assert_deploy_success(deploy_global_fixtures('globals', selector: 'test=prune2'))
```

Comment:
we should refute that the logs include "[pP]rune[d]"

Reply:
We use the test name to modify the fixture name, so if we refute 'prune' we can't use 'prune' in the test name. Instead I'm refuting 'pruned'.